### PR TITLE
Find only shared IE libraries, not plugins

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
In [this](https://github.com/openvinotoolkit/openvino/pull/3656) change some libraries like `inference_engine_ir_reader` become plugins and MUST NOT be found by OpenCV cmake since OpenCV tries to link them to `opencv_dnn`, while plugins (`.so` on OSX) cannot be linked to other targets.

### CODE REVIEW

File: cmake/OpenCVDetectInferenceEngine.cmake [hunk 1]
```diff
@@ -62,9 +62,7 @@ function(add_custom_ie_build _inc _lib _lib_rel _lib_dbg _msg)
     if(find_prefix STREQUAL "_empty_")  # foreach doesn't iterate over empty elements
       set(find_prefix "")
     endif()
-    foreach(find_suffix ${CMAKE_FIND_LIBRARY_SUFFIXES})
-      ocv_ie_find_extra_libraries("${find_prefix}" "${find_suffix}")
-    endforeach()
+    ocv_ie_find_extra_libraries("${find_prefix}" ".${CMAKE_SHARED_LIBRARY_SUFFIX}")
```
Comment (alalek) [inline cmake/OpenCVDetectInferenceEngine.cmake:65]: This variable already has dot prefix:
```
Linux: CMAKE_SHARED_LIBRARY_SUFFIX=.so
Mac: CMAKE_SHARED_LIBRARY_SUFFIX=.dylib
Windows: CMAKE_SHARED_LIBRARY_SUFFIX=.dll
```

On Windows it just wrong (must be `.lib`).

**Review by alalek — COMMENTED** [review]
This change doesn't look right.

Can we finally use `find_package(InferenceEngine) + find_package(ngraph)` instead of these hacks? (and remove these hacks completely)
What is still missing to support that?

File: cmake/OpenCVDetectInferenceEngine.cmake [hunk 2]
```diff
@@ -62,9 +62,7 @@ function(add_custom_ie_build _inc _lib _lib_rel _lib_dbg _msg)
     if(find_prefix STREQUAL "_empty_")  # foreach doesn't iterate over empty elements
       set(find_prefix "")
     endif()
-    foreach(find_suffix ${CMAKE_FIND_LIBRARY_SUFFIXES})
-      ocv_ie_find_extra_libraries("${find_prefix}" "${find_suffix}")
```
Comment (alalek) [inline cmake/OpenCVDetectInferenceEngine.cmake:66]:
```
if(NOT DEFINED INFERENCE_ENGINE_FIND_LIBRARY_SUFFIXES)  # allow custom override
  set(INFERENCE_ENGINE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
  if(APPLE)
    ocv_list_filterout(INFERENCE_ENGINE_FIND_LIBRARY_SUFFIXES "^.so$")  # skip plugins (can't be linked)
  endif()
endif()
foreach(find_prefix ${__prefixes})
  ...
  foreach(find_suffix ${INFERENCE_ENGINE_FIND_LIBRARY_SUFFIXES})
    ocv_ie_find_extra_libraries("${find_prefix}" "${find_suffix}")
  endforeach()
  ...
endforeach()
```

Comment (ilya-lavrenov) [reply cmake/OpenCVDetectInferenceEngine.cmake:66]: fixed.

File: cmake/OpenCVDetectInferenceEngine.cmake (hunk 1 above)
Comment (ilya-lavrenov) [reply cmake/OpenCVDetectInferenceEngine.cmake]: fixed.

**Comment by ilya-lavrenov** [comment]
> What is still missing to support that?

I suppose need to add a version to `InferenceEngineConfig.cmake`, but it's not a blocker.
Everything works with find_package on my machine. But CI still passed variables `INF_ENGINE_LIB_DIRS` which enables old school way to find IE inside OpenCV, while we need to pass path to `InferenceEngine_DIR`

CC @asuhov can we update CI variables to use find package?

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Pull request merged**